nextstrain/cli/remote/nextstrain_dot_org.py

```diff
@@ -0,0 +1,697 @@
+                    data = data, # type: ignore
+                    headers = {
+                        "Content-Type": media_type,
+                        "Content-Encoding": "gzip",
+                        **authorization })
+
+                raise_for_status(response)
+
+        # Upload datasets
+        for dataset, files in datasets.items():
+            for media_type, file in files.items():
+                endpoint = destination = api_endpoint(path if single else path / dataset)
+
+                if media_type != "application/vnd.nextstrain.dataset.main+json":
+                    destination += f" ({sidecar_suffix(media_type)})"
+
+                yield file, destination
+
+                put(endpoint, file, media_type)
+
+        # Upload narratives
+        for narrative, files in narratives.items():
+            markdown_type = "text/vnd.nextstrain.narrative+markdown"
+
+            assert set(files) == {markdown_type}, files
+
+            file = files[markdown_type]
+
+            if not narratives_only(path):
+                narrative = f"narratives/{narrative}"
+
+            endpoint = destination = api_endpoint(path if single else path / narrative)
+
+            yield file, destination
+
+            put(endpoint, file, markdown_type)
+
+
+def download(url: urllib.parse.ParseResult, local_path: Path, recursively: bool = False) -> Iterable[Tuple[str, Path]]:
+    """
+    Download the datasets or narratives deployed at the given remote *url*,
+    optionally *recursively*, saving them into the *local_dir*.
+    """
+    path = remote_path(url)
+
+    if str(path) == "/" and not recursively:
+        raise UserError(f"""
+            No path specified in URL ({url.geturl()}); nothing to download.
+
+            Did you mean to use --recursively?
+            """)
+
+    authorization = auth_headers()
+
+    with requests.Session() as http:
+        resources = _ls(path, recursively = recursively, http = http, authorization = authorization)
+
+        if not resources:
+            raise UserError(f"Path {path} does not seem to exist")
+
+        for resource in resources:
+            for subresource in resource.subresources:
+                # Remote source
+                endpoint = source = api_endpoint(resource.path)
+
+                if not subresource.primary:
+                    source += f" ({sidecar_suffix(subresource.media_type)})"
+
+                response = http.get(
+                    endpoint,
+                    headers = {"Accept": subresource.media_type, **authorization},
+                    stream = True)
+
+                with response:
+                    # Skip/ignore missing sidecars
+                    if response.status_code == 404 and not subresource.primary:
+                        continue
+
+                    # Check for bad response
+                    raise_for_status(response)
+                    assert content_media_type(response) == subresource.media_type
```

Comment:
I ran `nextstrain remote download https://nextstrain.org/ebola` locally and got this error:

```
% nextstrain remote download https://nextstrain.org/ebola                    
Traceback (most recent call last):
  File "/Users/vlin/Dropbox/repos/cli/nextstrain/cli/__init__.py", line 33, in run
    return opts.__command__.run(opts)
  File "/Users/vlin/Dropbox/repos/cli/nextstrain/cli/command/remote/download.py", line 89, in run
    for remote_file, local_file in downloads:
  File "/Users/vlin/Dropbox/repos/cli/nextstrain/cli/remote/nextstrain_dot_org.py", line 298, in download
    assert content_media_type(response) == subresource.media_type
AssertionError


An error occurred (see above) that likely indicates a bug in the
Nextstrain CLI.

To report this, please open a new issue and include the error above:
    <https://github.com/nextstrain/cli/issues/new/choose>
```

It looks like `content_media_type(response) = 'text/html'` and `subresource.media_type = 'application/vnd.nextstrain.dataset.main+json'`. Am I using this properly?

Reply:
Hmm. Did you start a local nextstrain.org server from the `restful-api` or `restful-api-rw` branch and set `NEXTSTRAIN_DOT_ORG=http://localhost:5000` before running `nextstrain remote`?

Reply:
Yes, using `restful-api-rw` branch. `nextstrain remote ls` works fine:

```
% echo "$NEXTSTRAIN_DOT_ORG"                             
http://localhost:5000
% nextstrain remote download https://nextstrain.org/ebola
'text/html'
'application/vnd.nextstrain.dataset.main+json'
Traceback (most recent call last):
  File "/Users/vlin/Dropbox/repos/cli/nextstrain/cli/__init__.py", line 33, in run
    return opts.__command__.run(opts)
  File "/Users/vlin/Dropbox/repos/cli/nextstrain/cli/command/remote/download.py", line 89, in run
    for remote_file, local_file in downloads:
  File "/Users/vlin/Dropbox/repos/cli/nextstrain/cli/remote/nextstrain_dot_org.py", line 300, in download
    assert content_media_type(response) == subresource.media_type
AssertionError


An error occurred (see above) that likely indicates a bug in the
Nextstrain CLI.

To report this, please open a new issue and include the error above:
    <https://github.com/nextstrain/cli/issues/new/choose>

% nextstrain remote ls https://nextstrain.org/     
https://nextstrain.org/dengue/denv1
https://nextstrain.org/dengue/denv2
https://nextstrain.org/dengue/denv3
https://nextstrain.org/dengue/denv4
...
```

Reply:
`https://nextstrain.org/ebola` works just fine for me. I suspect you didn't export `NEXTSTRAIN_DOT_ORG`, so it's visible to your shell (as the `echo` shows) but not to any processes your shell launches.

Ah, and actually your `nextstrain remote ls` confirms this: the output would show the overridden URL, e.g. `http://localhost:5000/dengue/denv1` and so on.

Reply:
Whoops that's it, works fine with `export`. False alarm!

Reply:
Oh good. :-) This also motivated me to update the instructions at the top to be a bit clearer.